container/container.go

```diff
@@ -90,9 +92,10 @@ type CommonContainer struct {
 	HasBeenStartedBefore   bool
 	HasBeenManuallyStopped bool // used for unless-stopped restart policy
 	MountPoints            map[string]*volume.MountPoint
-	HostConfig             *containertypes.HostConfig        `json:"-"` // do not serialize the host config in the json, otherwise we'll make the container unportable
-	ExecCommands           *exec.Store                       `json:"-"`
-	Secrets                []*containertypes.ContainerSecret `json:"-"` // do not serialize
+	HostConfig             *containertypes.HostConfig `json:"-"` // do not serialize the host config in the json, otherwise we'll make the container unportable
+	ExecCommands           *exec.Store                `json:"-"`
+	SecretStore            agentexec.SecretGetter     `json:"-"`
```

Comment:
While I agree with @stevvooe's assessment, I think this is not exactly what I had in mind after reading those comments.

For instance, swarmkit has a type defined in protobufs for mounts which is used by swarmkit (only), and then there's the API that docker exposes which is in api/types/mount.

I'm wondering too if the `github.com/docker/docker/api/types/swarm.Secret` type is even used now?

But in any case, I think this is the first time we've imported from swarmkit's actual agent.